### ISSUE TITLE
Version 0.17.8 broke Celery 

### ISSUE THREAD

**Issue by ur001** [getsentry/sentry-python#881]
I use Celery 4.4.7 with Django 1.11.29. Nothing unusual:
```python
os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'project_name.settings')
app = Celery('project_name')
app.config_from_object('django.conf:settings', namespace='CELERY')
app.autodiscover_tasks()
```

After updating from sentry-sdk from 0.17.7 to 0.17.8 celery breaks:
```
Unbound method apply_async() must be called with 
Task instance as first argument (got TaskType instance instead)
```

I think this is because of the following change in 0.17.8:
https://github.com/getsentry/sentry-python/commit/4bf4859087f2018f072fc0be472b7a12b58563e9#diff-bbcc458d29b0f1880cf429abbf8a94a5e8517f61072945d783e44fec4969a10b
```
@@ -60,9 +60,8 @@ class CeleryIntegration(Integration):
                 # Need to patch both methods because older celery sometimes
                 # short-circuits to task.run if it thinks it's safe.
                 task.__call__ = _wrap_task_call(task, task.__call__)
                 task.run = _wrap_task_call(task, task.run)
-                task.apply_async = _wrap_apply_async(task, task.apply_async)

                 # `build_tracer` is apparently called for every task
                 # invocation. Can't wrap every celery task for every invocation
                 # or we will get infinitely nested wrapper functions.
@@ -71,8 +70,12 @@ class CeleryIntegration(Integration):
             return _wrap_tracer(task, old_build_tracer(name, task, *args, **kwargs))

         trace.build_tracer = sentry_build_tracer

+        from celery.app.task import Task  # type: ignore
+
+        Task.apply_async = _wrap_apply_async(Task.apply_async)
+
```

**Comment by untitaker** [comment]
This is a duplicate of #844

**Comment by ur001** [comment]
@untitaker I'am not use custom Task class, tenant-schemas-celery, django-celery and other libraries for celery. I use pure Celery with pure Django with `@shared_task` or `@periodic_task` decorators and it brakes with sentry 0.17.8

**Comment by ur001** [comment]
@untitaker sory, I'm wrong. Not all tasks was broken! Only 2 cases:
1) Task with custom class (I forgot about one in my project)
2) Tasks with `@periodic_task` running manually with `taskname.delay(...)`. In fact, periodic tasks has custom class `PeriodicTask(Task)`

**Comment by untitaker** [comment]
ah ok cool, so it's probably going to be fixed as part ot #844 then. please subscribe to that task and ping me (or create a new issue) when you still encounter the issue after that one has been fixed
